### PR TITLE
JAMES-1871: Canceling JMAP uploads should not return an error

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/UploadServlet.java [hunk 1]
```diff
@@ -54,13 +55,23 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
         } else {
             try {
```
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/UploadServlet.java]: Then, you can extract the else closure.

**Comment by aduprat** [comment]
Have a tried to implement an integration test?

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/UploadServlet.java (hunk 1 above)
Comment (quynhn) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/UploadServlet.java]: OK, it's done

**Comment by quynhn** [comment]
```
Have a tried to implement an integration test?
```
I've tried to add more integration test but I can't. In fact, the cancelled upload as in the issues is user's action. I don't know how to make that action from integration test. I tried to set up connectionTimeout or socketTimeout but later on it cannot verify the response.
Could you show me how to do it?

**Comment by mbaechler** [comment]
It should not be that hard, client-side you retrieve the inputstream and fail after some bytes of reading.

**Comment by aduprat** [comment]
Here's a solution: #543

**Comment by quynhn** [comment]
It's cloned to https://github.com/linagora/james-project/pull/543

**Pull request closed without merging**